dev/io.openliberty.org.jboss.weld5/build.gradle

```diff
@@ -0,0 +1,161 @@
+
+        String templateFile = Files.readString(Paths.get("${pathToTemplateManifest}"));
+        templateFile = templateFile.replaceAll("#INSERTEXPORTS", exportsBndFormat).replaceAll("#INSERTIMPORTS", importsBndFormat)
+
+        try (PrintWriter out = new PrintWriter("${tempPathToNewManifest}")) {
+            out.print(templateFile)
+            out.flush()
+        }
+
+        // To avoid even touching the file if we do not need to, check we don't have an identical file already
+        if (! Files.exists(Paths.get(pathToNewManifest)) || Files.mismatch(Paths.get(tempPathToNewManifest), Paths.get(pathToNewManifest)) != -1L) {
+            //but if we spot a difference in an automated build we throw an exception
+            if (isAutomatedBuild) {
+                throw new GradleException('The generated bnd file did not mach the pre-existing bnd file. Did you check in check-me-in-if-i-change.bnd?')
+            }
+            Files.copy(Paths.get(tempPathToNewManifest), Paths.get(pathToNewManifest), StandardCopyOption.REPLACE_EXISTING)
+        }
+    }
+}
+
+//Valid inputs: "Export-Package:", "Import-Package:"
+ext.extractFromManifest = { sectionHeader ->
+    def returnArray = new ArrayList<String>()
+
+    //First put all the exports/imports into one big string.
+    String bigImportExportString = ""
+    try (BufferedReader br = new BufferedReader(new FileReader("${pathToOSManifest}"))) {
+        String line = br.readLine()
+        boolean foundSection = false
+        while (line != null) {
+            if (foundSection && !line.startsWith(" ")) { //The first line that does not start with a whitespace will be the end of a section
+                break
+            } else if (foundSection) {
+                bigImportExportString = bigImportExportString + line
+            } else if (line.contains(sectionHeader)) {
+                bigImportExportString = bigImportExportString + line.replaceAll(sectionHeader, "")
+                foundSection = true;
+            }
+            line = br.readLine()
+        }
+    }
+
+    //now remove all the whitespace and split into individual imports.
+    bigImportExportString=bigImportExportString.replaceAll("\s","")
+
+    int copyStartIndex = 0
+    boolean insideQuote = false
+
+    int index = 0
+    while (index < bigImportExportString.size()) {
+       char c = bigImportExportString.charAt(index)
+       if (c == '"') { 
+           insideQuote = !insideQuote
+       }
+       if (c == ',' && !insideQuote) {
+           returnArray.add(bigImportExportString.substring(copyStartIndex, index))
+           copyStartIndex = index+1 //This +1  skips over the ',' seperating two entries. As a seperator they're not part of either.
+       }
+       if (index == bigImportExportString.size() -1) {
+           returnArray.add(bigImportExportString.substring(copyStartIndex))
+       }
+       index++
+    }
+
+    return returnArray
+}
+
+compileJava {
+    dependsOn bndMagic
+}
```

Comment:
This doesn't run early enough for bnd to pick up the modified bnd file.

I'm not sure we _can_ run a gradle task early enough for bnd to see the changes, unfortunately.

Might be better as a manually run task.